test/extension-tests/successful-build/test/configure-and-build.test.ts

```diff
@@ -107,103 +113,125 @@ suite('Build', async () => {
   }).timeout(100000);
 
-  test('Test kit switch between different preferred generators and compilers', async function(this: ITestCallbackContext) {
-    // Select compiler build node dependent
-    const os_compilers : {[osName: string]: { kitLabel:RegExp, compiler:string}[]} = { ['linux']: [
-      {kitLabel: /^GCC 4.8.4/, compiler:'GNU GCC/G++' },
-      {kitLabel: /^Clang 5.0.0/, compiler:'GNU GCC/G++' }],
-      ['win32']: [
-      {kitLabel: /^GCC 4.8.1/, compiler:'GNU GCC/G++' },
-      {kitLabel: /^VisualStudio/, compiler:'Microsoft Visual Studio' }
-    ]};
-    if( !(workername in os_compilers)) this.skip();
-    const compiler = os_compilers[workername];
-
-    testEnv.kitSelection.defaultKitLabel=compiler[0].kitLabel;
-    await cmt.selectKit();
-    await cmt.build();
-
-
-    testEnv.kitSelection.defaultKitLabel=compiler[1].kitLabel;
-    await cmt.selectKit();
-    await cmt.build();
-
-    const result1 = await testEnv.result.getResultAsJson();
-    expect(result1['compiler']).to.eql(compiler[1].compiler);
-  }).timeout(100000);
-
-  test.only('Test kit switch between different preferred generators and same compiler', async function(this: ITestCallbackContext) {
-    // Select compiler build node dependent
-    const os_compilers : {[osName: string]: { kitLabel:RegExp, generator: string}[]} = { ['linux']: [
-      {kitLabel: /^Generator switch test GCC Make/, generator: 'Unix Makefiles' },
-      {kitLabel: /^Generator switch test GCC Ninja/, generator: 'Ninja' }],
-      ['win32']: [
-      {kitLabel: /^Generator switch test GCC Mingw - Win/, generator: 'MinGW Makefiles' },
-      {kitLabel: /^Generator switch test GCC Ninja - Win/, generator: 'Ninja' }
-    ]};
-    if( !(workername in os_compilers)) this.skip();
-    const compiler = os_compilers[workername];
-
-    testEnv.config.updatePartial({preferredGenerators: []});
-    testEnv.kitSelection.defaultKitLabel=compiler[0].kitLabel;
-    await cmt.selectKit();
-
-    await cmt.build();
-
-    testEnv.kitSelection.defaultKitLabel=compiler[1].kitLabel;
-    await cmt.selectKit();
-
-    await cmt.build();
-    const result1 = await testEnv.result.getResultAsJson();
-    expect(result1['cmake-generator']).to.eql(compiler[1].generator);
-  }).timeout(100000);
+  test('Test kit switch between different preferred generators and compilers',
+       async function(this: ITestCallbackContext) {
+         // Select compiler build node dependent
+         const os_compilers: {[osName: string]: {kitLabel: RegExp, compiler: string}[]} = {
+           linux: [
+             {kitLabel: /^GCC/, compiler: 'GNU GCC/G++'},
+             {kitLabel: /^Clang/, compiler: 'Clang/LLVM'},
+           ],
+           win32: [
+             {kitLabel: /^GCC/, compiler: 'GNU GCC/G++'},
+             {kitLabel: /^VisualStudio/, compiler: 'Microsoft Visual Studio'}
+           ]
+         };
+         this.skip();  // XXX: Fix race between setting kit and building
```

Comment:
should the test be disabled?

Reply:
This test seems unreliable, at least on my system.

Reply:
I am fine merging these disabled tests, although I'd prefer not to. I want to address the way kits are set in my other PR for 1.x.